File: doc_app/app/Models/UserDetails.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class UserDetails extends Model
{
    protected $fillable = [
        'user_id',
        'status'
        'bio_data',
        //'fav',
    ];

    public function user(){
        return $this->belongsTo(User::class);
    }
}

File: lib/models/userdetails.dart
class UserDetails {
  final int? id;
  final int userId;
  final String bioData;
  final String fav;
  final String status;

  UserDetails({
    this.id,
    required this.userId,
    required this.bioData,
    required this.fav,
    required this.status,
  });

  Map<String, dynamic> toMap() {
    return {
      'id': id,
      'user_id': userId,
      'bio_data': bioData,
      'fav': fav,
      'status': status,
    };
  }

  factory UserDetails.fromMap(Map<String, dynamic> map) {
    return UserDetails(
      id: map['id'],
      userId: map['user_id'],
      bioData: map['bio_data'],
      fav: map['fav'],
      status: map['status'],
    );
  }
}
